packages/block-editor/src/hooks/block-bindings.js

```diff
@@ -299,10 +299,10 @@ export const BlockBindingsPanel = ( { name: blockName, metadata } ) => {
 		setModalState( null );
 	};
 
+	const registeredSources = getBlockBindingsSources();
```

Comment:
Is it private and can't be accessed through `useSelect`?

Reply:
Seems that way: https://github.com/WordPress/gutenberg/blob/c5e8bd04fb30335829d7fad974423bcd05781a64/packages/blocks/src/api/registration.js#L1083-L1085. We might be able to `unlock` it here, though (?)

(Would you prefer using the selector?)

Reply:
In other words, I think the error comes from the fact that `getAllBlockBindingsSources` returns different results on subsequent executions and that's why `_sources` makes React think it's the same object even when the data changes.

Reply:
That should stop though if I move the call outside of the `useSelect`, and [add the resulting `registeredSources` as a param to `useSelect`](https://github.com/WordPress/gutenberg/pull/72109/files#diff-bff98e322c4269c50aa13e4b251beb9d53d2d0c3d4066406f735d2d49bed83a1R433), no? (Unfortunately, it didn't.)

Reply:
Hmm, the problem is that it's a different store (blocks), and `unlock` isn't available across package boundaries AFAICS 😕

Reply:
I think the main problem is that we need to run `source.editorUI()` (or `source.getFieldsList()`, respectively), and pass `context` and `select` to that function.

I have an idea how to untangle this, though. (In broad strokes, loop over `source`, and move the `useSelect` inside the loop.)